pkg/cmd/roachtest/roachtestutil/mixedversion/steps.go

```diff
@@ -715,3 +715,51 @@ func startStopOpts(opts ...option.StartStopOption) []option.StartStopOption {
 		option.NoBackupSchedule,
 	}, opts...)
 }
+
+type panicNodeStep struct {
+	rt         test.Test
+	initTarget int
+}
+
+func (s panicNodeStep) Background() shouldStop { return nil }
+
+func (s panicNodeStep) Description() string { return "run panic node" }
+
+func (s panicNodeStep) Run(ctx context.Context, l *logger.Logger, rng *rand.Rand, h *Helper) error {
+	nodeList := h.System.Descriptor.Nodes
+	targetNode := nodeList[rng.Intn(len(nodeList))]
+	nodeVersion, err := h.System.NodeVersion(targetNode)
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to get node version from target node %d", targetNode)
+	}
+
+	binaryPath := clusterupgrade.CockroachPathForVersion(s.rt, nodeVersion)
+	settings := install.MakeClusterSettings(
+		install.BinaryOption(binaryPath),
+		install.TagOption(systemTag),
+	)
+	customStartOpts := []option.StartStopOption{option.WithInitTarget(s.initTarget)}
+
+	h.ExpectDeath()
+
+	const stmt = "SELECT crdb_internal.force_panic('expected panic from panicNodeMutator')"
+	if err = h.System.ExecWithGateway(
+		rng,
+		option.NodeListOption{targetNode},
+		stmt,
+	); err == nil {
+		return errors.New("expected connection failure due to node panic, but got nil")
+	}
+
+	startCtx, cancel := context.WithTimeout(ctx, startTimeout)
```

Comment:
We probably want some sort of random wait time between restarting the node. `server.time_until_store_dead` is one duration of interest. By default this is `5m`, and it represents how long the cluster will wait until it declares a store dead and starts rebalancing it's ranges.

Some wait times fixated around that could be a good idea, e.g. [0m, 3m, 6m, 10m] would give us: no waiting at all, some waiting but no rebalancing, rebalancing just starting, and rebalancing almost done/done.

You do want to take special care to not wait too long for local runs though. These might be run in CI so we definitely don't want to block for 10 minutes, or maybe at all.